plugins/inputs/ipset/ipset.go

```diff
@@ -56,6 +58,14 @@ func (i *Ipset) Gather(acc telegraf.Accumulator) error {
 	scanner := bufio.NewScanner(out)
 	for scanner.Scan() {
 		line := scanner.Text()
+
+		if i.CountPerIPEntries {
+			err := i.entriesParser.addLine(line, acc)
+			if err != nil {
+				acc.AddError(err)
+			}
```

Comment:
The `AddError` function already checks for the error being `nil`...
```suggestion
			acc.AddError(i.entriesParser.addLine(line, acc))
```

Reply:
oh, ok